zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/builder/component/GroupBuilder.kt

```diff
@@ -4,19 +4,20 @@ import org.hexworks.zircon.api.builder.Builder
 import org.hexworks.zircon.api.component.ColorTheme
 import org.hexworks.zircon.api.component.Component
 import org.hexworks.zircon.api.component.Group
-import org.hexworks.zircon.internal.dsl.ZirconDsl
 import org.hexworks.zircon.api.resource.TilesetResource
 import org.hexworks.zircon.internal.component.impl.DefaultGroup
 import org.hexworks.zircon.internal.config.RuntimeConfig
+import org.hexworks.zircon.internal.dsl.ZirconDsl
 import kotlin.jvm.JvmStatic
 
 @ZirconDsl
-class GroupBuilder<T : Component>(
-    private var isDisabled: Boolean = false,
-    private var isHidden: Boolean = false,
-    private var theme: ColorTheme = RuntimeConfig.config.defaultColorTheme,
-    private var tileset: TilesetResource = RuntimeConfig.config.defaultTileset
-) : Builder<Group<T>> {
+class GroupBuilder<T : Component> : Builder<Group<T>> {
+
+    var isDisabled: Boolean = false
+    var isHidden: Boolean = false
+    var theme: ColorTheme = RuntimeConfig.config.defaultColorTheme
+    var tileset: TilesetResource = RuntimeConfig.config.defaultTileset
+    var name: String = ""
 
```

Comment:
this is very convenient for the dsl but what do you think about it polluting the builder's public api? though i think it may be a fair tradeoff

Reply:
applies for all cases in PR where public vars are introduced

Reply:
I think the `with*` functions can be removed later if we abandon Java support 😛

Reply:
hah, i like the sound of that.. then all these builders could be renamed to Scopes and their constructor made internal or something, dsl all the way :P

Reply:
What is a Scope in this context? I'm maybe not that familiar.

Reply:
a Scope is generally the class who's scope is opened by a dsl element, basically the receiver component of a dsl.

Reply:
Where can I read up on this? I think this is the first time I hear about it.

Reply:
Well, I went to dig up the one example I remembered and found that it did not call its class LayoutScope, but LayoutBuilder. LOL, it may be something that I came up with myself.. Either way it's a purely semantic argument, the function of the class remains exactly he same in this contents, I think I just decided to call them Scopes when writing a dsl because they were in practice only surfacing as specific scopes exposed by the dsl. 
I'm happy with them staying named Builders, only worry that if we take away their ability to be used according to the Builder pattern that may confuse people.

Reply:
A rename is not feasible because it would break the API. They can stay this way though, since we still support Java. We'll see how this evolves.